File: dmn-core/src/test/resources/tck/cl3/expected/dmn/0017-tabletests/type/TA.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinitionInterface.ftl", "tA"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
public interface TA extends com.gs.dmn.runtime.DMNType {
    static TA toTA(Object other) {
        if (other == null) {
            return null;
        } else if (TA.class.isAssignableFrom(other.getClass())) {
            return (TA)other;
        } else if (other instanceof com.gs.dmn.runtime.Context) {
            TAImpl result_ = new TAImpl();
            result_.setName((String)((com.gs.dmn.runtime.Context)other).get("name"));
            result_.setPrice((java.math.BigDecimal)((com.gs.dmn.runtime.Context)other).get("price"));
            return result_;
        } else if (other instanceof com.gs.dmn.runtime.DMNType) {
            return toTA(((com.gs.dmn.runtime.DMNType)other).toContext());
        } else {
            throw new com.gs.dmn.runtime.DMNRuntimeException(String.format("Cannot convert '%s' to '%s'", other.getClass().getSimpleName(), TA.class.getSimpleName()));
        }
    }

    @com.fasterxml.jackson.annotation.JsonGetter("name")
    String getName();

    @com.fasterxml.jackson.annotation.JsonGetter("price")
    java.math.BigDecimal getPrice();

    default com.gs.dmn.runtime.Context toContext() {
        com.gs.dmn.runtime.Context context = new com.gs.dmn.runtime.Context();
        context.put("name", getName());
        context.put("price", getPrice());
        return context;
    }

    default boolean equalTo(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        TA other = (TA) o;
        if (this.getName() != null ? !this.getName().equals(other.getName()) : other.getName() != null) return false;
        if (this.getPrice() != null ? !this.getPrice().equals(other.getPrice()) : other.getPrice() != null) return false;

        return true;
    }

    default int hash() {
        int result = 0;
        result = 31 * result + (this.getName() != null ? this.getName().hashCode() : 0);
        result = 31 * result + (this.getPrice() != null ? this.getPrice().hashCode() : 0);
        return result;
    }

    default String asString() {
        StringBuilder result_ = new StringBuilder("{");
        result_.append("name=" + getName());
        result_.append(", price=" + getPrice());
        result_.append("}");
        return result_.toString();
    }
}

File: dmn-test-cases/standard/tck/1.1/cl3/0017-tableTests/translator/expected/kotlin/dmn/type/TA.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinitionInterface.ftl", "tA"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(`as` = type.TAImpl::class)
interface TA : com.gs.dmn.runtime.DMNType {
    @get:com.fasterxml.jackson.annotation.JsonGetter("name")
    val name: String?

    @get:com.fasterxml.jackson.annotation.JsonGetter("price")
    val price: java.math.BigDecimal?

    override fun toContext(): com.gs.dmn.runtime.Context {
        val context = com.gs.dmn.runtime.Context()
        context.put("name", this.name)
        context.put("price", this.price)
        return context
    }

    fun equalTo(o: Any?): Boolean {
        if (this === o) return true
        if (javaClass != o?.javaClass) return false

        val other = o as TA
        if (if (this.name != null) this.name != other.name else other.name != null) return false
        if (if (this.price != null) this.price != other.price else other.price != null) return false

        return true
    }

    fun hash(): Int {
        var result = 0
        result = 31 * result + (if (this.name != null) this.name.hashCode() else 0)
        result = 31 * result + (if (this.price != null) this.price.hashCode() else 0)
        return result
    }

    fun asString(): String {
        val result_ = StringBuilder("{")
        result_.append("name=" + name)
        result_.append(", price=" + price)
        result_.append("}")
        return result_.toString()
    }

    companion object {
        @JvmStatic
        fun toTA(other: Any?): TA? {
            if (other == null) {
                return null
            } else if (other is TA?) {
                return other
            } else if (other is com.gs.dmn.runtime.Context) {
                var result_ = TAImpl()
                result_.name = other.get("name") as String?
                result_.price = other.get("price") as java.math.BigDecimal?
                return result_
            } else if (other is com.gs.dmn.runtime.DMNType) {
                return toTA(other.toContext())
            } else {
                throw com.gs.dmn.runtime.DMNRuntimeException(String.format("Cannot convert '%s' to '%s'", other.javaClass.getSimpleName(), TA::class.java.getSimpleName()))
            }
        }
    }
}
